app.py

```diff
@@ -34,9 +34,38 @@ def channel(): #This is the ChannelPage we will send variabls and stuff here to
     return render_template("ChannelPage.html", channels=channels)
 
 
-@app.route('/profile')
-def profile(): #This is the profile page we will send variables and stuff here to configure
-    return render_template("ProfilePage.html")
+@app.route('/Profile')
+def profile():  # This is the profile page we will send variables and stuff here to configure
+    username = 'imoudu'
+
+    user_information = db.retrieve_from_user(username)
+
+    return render_template("ProfilePage.html", userInformation=user_information)
+
+
+# This function handles
+@app.route('/Update', methods=['POST', 'GET'])
+def update():
+    original = request.form['original_username']
+
+    # get the original information
+    user_information = db.retrieve_from_user(original)
+
+    # the html form is formatted in a way such that the default text is the original information
+    # So if nothing was changed then all the information will remain the same
+    user_information["username"] = request.form['username']
+    user_information["first"] = request.form['first']
+    user_information["last"] = request.form['last']
+    user_information["email"] = request.form['email']
+    user_information["password"] = request.form['password']
+
+    # create the tuple for updating
+    # "UPDATE User SET username = ?, firstname = ?, lastname = ?, email = ?, password = ? WHERE username  = ?"
```

Comment:
With the new database that we are using we will have to change this. See the channel() function in app.py to see how to do this. Simple fix but wont work anymore if not made.